plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3ClientSettings.java

```diff
@@ -172,6 +172,48 @@ final class S3ClientSettings {
         key -> Setting.timeSetting(key, TimeValue.timeValueMillis(50_000), Property.NodeScope)
     );
 
+    /** The request timeout for connecting to s3. */
+    static final Setting.AffixSetting<TimeValue> REQUEST_TIMEOUT_SETTING = Setting.affixKeySetting(
+        PREFIX,
+        "request_timeout",
+        key -> Setting.timeSetting(key, TimeValue.timeValueMinutes(2), Property.NodeScope)
+    );
+
+    /** The connection timeout for connecting to s3. */
+    static final Setting.AffixSetting<TimeValue> CONNECTION_TIMEOUT_SETTING = Setting.affixKeySetting(
+        PREFIX,
+        "connection_timeout",
+        key -> Setting.timeSetting(key, TimeValue.timeValueSeconds(10), Property.NodeScope)
+    );
+
+    /** The connection TTL for connecting to s3. */
+    static final Setting.AffixSetting<TimeValue> CONNECTION_TTL_SETTING = Setting.affixKeySetting(
+        PREFIX,
+        "connection_ttl",
+        key -> Setting.timeSetting(key, TimeValue.timeValueMillis(5000), Property.NodeScope)
+    );
+
+    /** The maximum connections to s3. */
+    static final Setting.AffixSetting<Integer> MAX_CONNECTIONS_SETTING = Setting.affixKeySetting(
+        PREFIX,
+        "max_connections",
+        key -> Setting.intSetting(key, 100, Property.NodeScope)
+    );
+
+    /** Connection acquisition timeout for new connections to S3. */
+    static final Setting.AffixSetting<TimeValue> CONNECTION_ACQUISITION_TIMEOUT = Setting.affixKeySetting(
+        PREFIX,
+        "connection_acquisition_timeout",
+        key -> Setting.timeSetting(key, TimeValue.timeValueMinutes(2), Property.NodeScope)
+    );
+
+    /** The maximum pending connections to S3. */
+    static final Setting.AffixSetting<Integer> MAX_PENDING_CONNECTION_ACQUIRES = Setting.affixKeySetting(
+        PREFIX,
+        "max_pending_connection_acquires",
+        key -> Setting.intSetting(key, 10_000, Property.NodeScope)
+    );
+
```

Comment:
Are these the SDK defaults?

Reply:
We will need to identify settings which we want to override .